x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -106,55 +106,82 @@ class AgentPolicyService {
     esClient: ElasticsearchClient
   ): Promise<{
     created: boolean;
-    defaultAgentPolicy: AgentPolicy;
+    policy: AgentPolicy;
   }> {
-    const agentPolicies = await soClient.find<AgentPolicySOAttributes>({
-      type: AGENT_POLICY_SAVED_OBJECT_TYPE,
+    const searchParams = {
       searchFields: ['is_default'],
       search: 'true',
-    });
+    };
+    return await this.ensureAgentPolicy(soClient, esClient, DEFAULT_AGENT_POLICY, searchParams);
+  }
 
-    if (agentPolicies.total === 0) {
-      const newDefaultAgentPolicy: NewAgentPolicy = {
-        ...DEFAULT_AGENT_POLICY,
-      };
+  public async ensureDefaultFleetServerAgentPolicy(
+    soClient: SavedObjectsClientContract,
+    esClient: ElasticsearchClient
+  ): Promise<{
+    created: boolean;
+    policy: AgentPolicy;
+  }> {
+    const searchParams = {
+      searchFields: ['is_default_fleet_server'],
+      search: 'true',
+    };
+    return await this.ensureAgentPolicy(
+      soClient,
+      esClient,
+      DEFAULT_FLEET_SERVER_AGENT_POLICY,
+      searchParams
+    );
+  }
 
-      return {
-        created: true,
-        defaultAgentPolicy: await this.create(soClient, esClient, newDefaultAgentPolicy),
-      };
-    }
+  public async ensurePreconfiguredAgentPolicy(
+    soClient: SavedObjectsClientContract,
+    esClient: ElasticsearchClient,
+    config: Omit<NewAgentPolicy, 'namespace'> & { namespace?: string }
```

Comment:
if you are agreeable to my suggestion of having a new `PreconfiguredAgentPolicy`, that can be used here instead